app/Services/AdventuresLeagueAdapter.php

```diff
@@ -105,39 +104,14 @@ private function getEntries($data, $characterId)
                 } else {
                     $type = Entry::TYPE_DOWNTIME;
                 }
-                $entryData = [
-                    'user_id' => Auth::id(),
-                    'adventure_id' => null,
-                    'campaign_id' => null,
-                    'character_id' => $characterId,
-                    'event_id' => null,
-                    'dungeon_master_id' => null,
-                    'dungeon_master' => (array_key_exists(12, $data[$i])) ? (float)$data[$i][12] : "",
-                    'location' => (array_key_exists(11, $data[$i])) ? (float)$data[$i][11] : "",
-                    'levels' => 0,
-                    'date_played' => ($data[$i][3] == "") ? now() : $data[$i][3],
-                    'gp' => (array_key_exists(7, $data[$i])) ? (float)$data[$i][7] : 0,
-                    'downtime' => (array_key_exists(8, $data[$i])) ? (float)$data[$i][8] : 0,
-                    'type' => $type,
-                ];
+                $entryData = $this->getEntryData($characterId, $data[$i], $type);
                 $entry = Entry::create($entryData);
                 $entry->save();
                 array_push($entries, $entry);
             } elseif ($isItemEntry) {
                 if (count($entries) == 0) {
-                    $e = Entry::create([
-                        'user_id' => Auth::id(),
-                        'adventure_id' => null,
-                        'campaign_id' => null,
-                        'character_id' => $characterId,
-                        'event_id' => null,
-                        'dungeon_master_id' => null,
-                        'levels' => 0,
-                        'date_played' => now(),
-                        'gp' => 0,
-                        'downtime' => 0,
-                        'type' => Entry::TYPE_GAME,
-                    ]);
+                    $entryData = $this->getEntryData($characterId, null, null);
```

Comment:
if you set defaults in the evoked function signature this can just be 
```suggestion
                    $entryData = $this->getEntryData($characterId);
```

Reply:
I never knew PHP had that feature! Thank you for mentioning it.

Reply:
no problem :)

Reply:
great suggestion, added!